src/twisted/protocols/basic.py

```diff
@@ -481,7 +481,7 @@ def lineLengthExceeded(self, line):
         Called when the maximum line length has been reached.
         Override if it needs to be dealt with in some special way.
         """
-        return error.ConnectionLost('Line length exceeded')
+        return self.transport.loseConnection()
```

Comment:
The "return" here is redundant, and somewhat misleading.

Reply:
Agreed.  Unfortunately [`LineReceiver` does this as well](https://github.com/requests/requests/issues/4006#issuecomment-300228176).  I'd rather deal with both in a ticket that merges their implementations, or better yet one that finally eliminates the long-deprecated return value of `dataReceived`.